File: core/core-domain/src/main/java/com/coffee/gu/stamp/StampCouponUsageRepository.java
package com.coffee.gu.stamp;

import java.util.List;

public interface StampCouponUsageRepository {
    StampCouponUsage save(StampCouponUsage stampCouponUsage);
    List<StampCouponUsage> saveAll(List<StampCouponUsage> stampCouponUsages);
    List<StampCouponUsage> findAllByStampIdIn(List<Long> stampIds);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/stamp/StampCouponUsageRepository.kt
package com.coffee.gu.stamp

interface StampCouponUsageRepository {
    fun save(stampCouponUsage: StampCouponUsage): StampCouponUsage
    fun saveAll(stampCouponUsages: List<StampCouponUsage>): List<StampCouponUsage>
    fun findAllByStampIdIn(stampIds: List<Long>): List<StampCouponUsage>
}
